src/main/flight/pid.c

```diff
@@ -967,9 +967,13 @@ void FAST_CODE pidController(float dT)
 
         // Limit desired rate to something gyro can measure reliably
         pidState[axis].rateTarget = constrainf(rateTarget, -GYRO_SATURATION_LIMIT, +GYRO_SATURATION_LIMIT);
+
 #ifdef USE_GYRO_KALMAN
-        gyroKalmanSetSetpoint(axis, pidState[axis].rateTarget);
+        if (gyroConfig()->kalmanEnabled) {
+            pidState[axis].gyroRate = gyroKalmanUpdate(axis, pidState[axis].gyroRate, pidState[axis].rateTarget);
```

Comment:
We are setting `gyroRate` twice if we are using kalman. This is redundant.

Reply:
actually not so much. Kalman needs setpoint and measurement and if it's enabled, it overrides the pre-kalman value